src/main.rs

```diff
@@ -127,32 +222,27 @@ fn main() {
 
         if let Some(endpoint) = usb_endpoint {
             let transport = match endpoint {
-                usb_ctap_hid::UsbEndpoint::MainHid => Transport::MainHid,
+                UsbEndpoint::MainHid => Transport::MainHid,
                 #[cfg(feature = "vendor_hid")]
-                usb_ctap_hid::UsbEndpoint::VendorHid => Transport::VendorHid,
+                UsbEndpoint::VendorHid => Transport::VendorHid,
             };
             let reply = ctap.process_hid_packet(&pkt_request, transport, now);
-            // This block handles sending packets.
-            for mut pkt_reply in reply {
-                let hid_connection = transport.hid_connection(ctap.env());
-                match hid_connection.send_or_recv_with_timeout(&mut pkt_reply, SEND_TIMEOUT) {
-                    Ok(SendOrRecvStatus::Timeout) => {
-                        #[cfg(feature = "debug_ctap")]
-                        print_packet_notice("Sending packet timed out", &clock);
-                        // TODO: reset the ctap_hid state.
-                        // Since sending the packet timed out, we cancel this reply.
+            if reply.has_data() {
+                // Update endpoint with the reply.
+                for ep in replies.replies.iter_mut() {
+                    if ep.endpoint == endpoint {
+                        if ep.reply.has_data() {
+                            #[cfg(feature = "debug_ctap")]
+                            writeln!(
+                                Console::new(),
+                                "Warning overwritting existing reply for endpoint {}",
```

Comment:
"Warning overwritting" is probably a typo.

Reply:
Fixed.